deso/events.go

```diff
@@ -44,63 +45,72 @@ func (node *Node) handleSnapshotCompleted() {
+				currentBlockHeight := uint64(1)
+				// We'll force a ceiling on this because otherwise the last block could amass O(snapshotBlockHeight) balances
+				balancesPerBlock := totalCount / snapshotBlockHeight
+				balancesMap := make(map[lib.PublicKey]uint64)
+				if totalCount < snapshotBlockHeight {
+					balancesPerBlock = 1
+				}
+				currentCounter := uint64(0)
+
+				for nodeIterator.Seek(prefix); nodeIterator.ValidForPrefix(prefix); nodeIterator.Next() {
+					key := nodeIterator.Item().Key()
+					keyCopy := make([]byte, len(key))
+					copy(keyCopy[:], key[:])
+
+					valCopy, err := nodeIterator.Item().ValueCopy(nil)
+					if err != nil {
+						return errors.Wrapf(err, "Problem iterating over chain database, "+
+							"on key (%v) and value (%v)", keyCopy, valCopy)
 					}
-					currentCounter := uint64(0)
-
-					for nodeIterator.Seek(prefix); nodeIterator.ValidForPrefix(prefix); nodeIterator.Next() {
-						key := nodeIterator.Item().Key()
-						keyCopy := make([]byte, len(key))
-						copy(keyCopy[:], key[:])
-
-						valCopy, err := nodeIterator.Item().ValueCopy(nil)
-						if err != nil {
-							return errors.Wrapf(err, "Problem iterating over chain database, "+
-								"on key (%v) and value (%v)", keyCopy, valCopy)
-						}
 
-						balance := lib.DecodeUint64(valCopy)
-						pubKey := lib.NewPublicKey(key[1:])
-						balancesMap[*pubKey] = balance
+					balance := lib.DecodeUint64(valCopy)
+					pubKey := lib.NewPublicKey(key[1:])
+					balancesMap[*pubKey] = balance
 
-						if err := node.Index.PutSingleBalanceSnapshotWithTxn(
-							indexTxn, currentBlockHeight, DESOBalance, *pubKey, balance); err != nil {
-							return errors.Wrapf(err, "Problem updating balance snapshot in index, "+
-								"on key (%v), value (%v), and height (%v)", keyCopy, valCopy,
-								snapshot.CurrentEpochSnapshotMetadata.FirstSnapshotBlockHeight)
-						}
+					if err := node.Index.PutSingleBalanceSnapshotWithWB(
+						wb, currentBlockHeight, DESOBalance, *pubKey, balance); err != nil {
+						return errors.Wrapf(err, "Problem updating balance snapshot in index, "+
+							"on key (%v), value (%v), and height (%v)", keyCopy, valCopy,
+							snapshot.CurrentEpochSnapshotMetadata.FirstSnapshotBlockHeight)
+					}
 
-						currentCounter += 1
-						if currentCounter >= balancesPerBlock && currentBlockHeight < snapshotBlockHeight {
-							node.Index.PutHypersyncBlockBalances(currentBlockHeight, DESOBalance, balancesMap)
-							balancesMap = make(map[lib.PublicKey]uint64)
-							currentBlockHeight++
-							currentCounter = 0
+					currentCounter += 1
+					if currentCounter >= balancesPerBlock && currentBlockHeight < snapshotBlockHeight {
+						node.Index.PutHypersyncBlockBalancesWithWB(wb, currentBlockHeight, DESOBalance, balancesMap)
+						balancesMap = make(map[lib.PublicKey]uint64)
+						currentBlockHeight++
+						currentCounter = 0
+						if currentBlockHeight%1000 == 0 {
```

Comment:
Let's pull this `1000` out into a variable outside of all the inner functoins so it's defined once and we can easily tweak it for performance.